datafusion/expr/src/logical_plan/plan.rs

```diff
@@ -1058,7 +1060,7 @@ pub struct CreateCatalogSchema {
 #[derive(Clone)]
 pub struct DropTable {
     /// The table name
-    pub name: String,
+    pub name: OwnedTableReference,
```

Comment:
This is the key change -- previously in DataFusion relation (table, view, etc) were stored as String which was then "parsed" (really split on `.`). This PR changes the code so the original parsed multi-part TableReference is used instead